### PR TITLE
Enable LAA1002 rule

### LINKED ISSUE #164: Turn on and fix LAA1002 errors to ensure actions are determnistic
> I temporarily turned off LAA1002 in this pull request to merge it.  This should be turned on again.

I temporarily suppressed LAA1002 errors in <https://github.com/planetarium/lib9c/pull/138#issuecomment-712550863>, but these errors should be properly addressed.

See also the commit 26448fbbd857fa8a1b24d22ed0dad2006aa3ebf7.

### PR DESCRIPTION
**Pull request by ipdae**
- resolve #164

### CODE REVIEW

File: Lib9c/Model/State/GameConfigState.cs [hunk 1]
```diff
@@ -65,12 +65,16 @@ public override IValue Serialize()
...
             };
+#pragma warning disable LAA1002
             return new Dictionary(values.Union((Dictionary) base.Serialize()));
+#pragma warning restore LAA1002
         }
 
         public void Set(GameConfigSheet sheet)
         {
+#pragma warning disable LAA1002
             foreach (var row in sheet)
```
Comment (longfin) [inline Lib9c/Model/State/GameConfigState.cs:76]: Is it okay that we don't care iteration order of `sheet`?

File: Lib9c/Model/State/WeeklyArenaState.cs [hunk 2]
```diff
@@ -222,7 +226,9 @@ public Address[] GetAgentAddresses(int count)
                     break;
             }
 
+#pragma warning disable LAA1002
```
Comment (longfin) [inline Lib9c/Model/State/WeeklyArenaState.cs:229]: Is it okay we don't mind the order of addresses?

File: Lib9c/TableData/Item/EquipmentItemSetEffectSheet.cs [hunk 3]
```diff
@@ -88,7 +89,8 @@ public static List<EquipmentItemSetEffectSheet.Row> GetSetEffectRows(this Equipm
                     rows.Add(row);
```
Comment (longfin) [inline Lib9c/TableData/Item/EquipmentItemSetEffectSheet.cs:89]: Is it okay we don't mind the order of `rows`?

File: Lib9c/TableData/Sheet.cs [hunk 4]
```diff
@@ -119,7 +119,9 @@ public void Set(string csv, bool isReversed = false)
 
         public void Set<T>(Sheet<TKey, T> sheet, bool executePostSet = true) where T : TValue, new()
         {
+#pragma warning disable LAA1002
             foreach (var sheetRow in sheet)
```
Comment (longfin) [inline Lib9c/TableData/Sheet.cs:123]: Is it okay we don't mind the order of `AddRow()`?

File: Lib9c/Model/State/GameConfigState.cs (hunk 1 above)
Comment (ipdae) [reply Lib9c/Model/State/GameConfigState.cs:76]: I don't think it's a problem. but, I'll change it to use `OrderedList`.

File: Lib9c/Model/State/WeeklyArenaState.cs (hunk 2 above)
Comment (ipdae) [reply Lib9c/Model/State/WeeklyArenaState.cs:229]: I don't think it's a problem. but, I'll change it.

File: Lib9c/Model/Character/CharacterBase.cs [hunk 5]
```diff
@@ -103,7 +103,9 @@ protected CharacterBase(CharacterBase value)
             Skills = value.Skills;
             // 버프는 컨테이너도 옮기고,
             Buffs = new Dictionary<int, Buff.Buff>();
+#pragma warning disable LAA1002
             foreach (var pair in value.Buffs)
```
Comment (dahlia) [inline Lib9c/Model/Character/CharacterBase.cs:107]: How about turning these lines into this?  This must pass the LAA1002 rule without suppressing it.

```csharp
Buffs = value.Buffs.ToDictionary(kv => kv.Key, kv => (Buff.Buff) kv.Value.Clone());
```

File: Lib9c/TableData/Item/EquipmentItemSetEffectSheet.cs (hunk 3 above)
Comment (ipdae) [reply Lib9c/TableData/Item/EquipmentItemSetEffectSheet.cs:89]: yes.

File: Lib9c/TableData/Sheet.cs (hunk 4 above)
Comment (ipdae) [reply Lib9c/TableData/Sheet.cs:123]: yes, beacuse sorting in `PostSet`.

https://github.com/planetarium/lib9c/blob/9fa307cd4cd9fbd4496ee50c714b703c380184c9/Lib9c/TableData/Sheet.cs#L160-L170

File: Lib9c/Model/Character/CharacterBase.cs [hunk 6]
```diff
@@ -391,7 +395,9 @@ public int GetCooldown(int skillId)
 
         public void ReduceCooldown()
         {
+#pragma warning disable LAA1002
             if (!_skillsCooldown.Any())
+#pragma warning restore LAA1002
```
Comment (dahlia) [inline Lib9c/Model/Character/CharacterBase.cs:400]: :+1: I am going to fix this false alarm: <https://github.com/planetarium/libplanet/issues/1048>.

File: Lib9c/Model/CollectionMap.cs [hunk 7]
```diff
@@ -53,7 +53,9 @@ public void Clear()
 
         public bool Contains(KeyValuePair<int, int> item)
         {
+#pragma warning disable LAA1002
             return _dictionary.Contains(item);
+#pragma warning restore LAA1002
```
Comment (dahlia) [inline Lib9c/Model/CollectionMap.cs:58]: :+1: I am going to fix this false alarm: <https://github.com/planetarium/libplanet/issues/1048>.

File: Lib9c/Model/State/WeeklyArenaState.cs (hunk 2 above)
Comment (ipdae) [reply Lib9c/Model/State/WeeklyArenaState.cs:229]: i found there is no using this method. i'll delete this method

Comment (longfin) [reply Lib9c/Model/State/WeeklyArenaState.cs:229]: 🆗

**Review by longfin — APPROVED** [review]

**Review by dahlia — APPROVED** [review]

**Pull request merged**